test-k8s.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+set -e
+
+export TERM=xterm-256color
+
+PROJECT_ROOT="$(pwd)"
+TEST_ROOT="${PROJECT_ROOT}/tests/integration"
+TIMEOUT_SECONDS=180
+RUN_ID="$(date +%y%m%d_%H%M%S)"
+LOG_DIR="${PROJECT_ROOT}/logs/${RUN_ID}"
+while [ -d "$LOG_DIR" ]; do
+    tmp_id="$((tmp_id + 1))"
+    new_run="${RUN_ID}_${tmp_id}"
+    LOG_DIR="${PROJECT_ROOT}/logs/${RUN_ID}"
+done
+RUN_ID=''
+
+mkdir -p "${PROJECT_ROOT}/logs"
+mkdir "$LOG_DIR"
+
+BUILD_LOG="${LOG_DIR}/build.log"
+RUN_LOG="${LOG_DIR}/run_master.log"
+TEST_LOG="${LOG_DIR}/test.log"
+WORKER_LOG_DIR="/tmp/jasminegraph"
+rm -rf "${WORKER_LOG_DIR}"
+mkdir -p "${WORKER_LOG_DIR}"
+
+build_and_run_on_k8s() {
+    cd "$PROJECT_ROOT"
+    docker build -t jasminegraph . |& tee "$BUILD_LOG"
+    build_status="${PIPESTATUS[0]}"
+    if [ "$build_status" != '0' ]; then
+        set +ex
+        echo
+        echo -e '\e[31;1mERROR: Build failed\e[0m'
+        rm -rf "${TEST_ROOT}/env"
+        exit "$build_status"
+    fi
+
+    set +e
+    clear_resources >/dev/null 2>&1
+    set -e
+
+    metadb_path="${TEST_ROOT}/env/databases/metadb" \
+        performancedb_path="${TEST_ROOT}/env/databases/performancedb" \
+        data_path="${TEST_ROOT}/env/data" \
+        log_path="${LOG_DIR}" \
+        envsubst <"${PROJECT_ROOT}/k8s/volumes.yaml" | kubectl apply -f -
```

Comment:
Shouldn't we require a file path here (`kubectl apply -f -`) as well since we are specifying `-f` ? Why do we have a `-` at the end of this line?

Reply:
the `-` says to read the stdin (which is piped from the other command). Similar to `-f -` in many other commands like `tar` and `grep`

Reply:
Ok, got it